libraries/chain/asset_evaluator.cpp

```diff
@@ -360,42 +360,60 @@ void_result asset_update_issuer_evaluator::do_apply(const asset_update_issuer_op
    return void_result();
 } FC_CAPTURE_AND_RETHROW( (o) ) }
 
-void_result asset_update_bitasset_evaluator::do_evaluate(const asset_update_bitasset_operation& o)
+void_result asset_update_bitasset_evaluator::do_evaluate(const asset_update_bitasset_operation& op)
 { try {
    database& d = db();
 
-   const asset_object& a = o.asset_to_update(d);
+   const asset_object& asset_obj = op.asset_to_update(d);
 
-   FC_ASSERT(a.is_market_issued(), "Cannot update BitAsset-specific settings on a non-BitAsset.");
+   FC_ASSERT(asset_obj.is_market_issued(), "Cannot update BitAsset-specific settings on a non-BitAsset.");
 
-   const asset_bitasset_data_object& b = a.bitasset_data(d);
-   FC_ASSERT( !b.has_settlement(), "Cannot update a bitasset after a settlement has executed" );
-   if( o.new_options.short_backing_asset != b.options.short_backing_asset )
+   const asset_bitasset_data_object& current_bitasset_data = asset_obj.bitasset_data(d);
+
+   FC_ASSERT( !current_bitasset_data.has_settlement(), "Cannot update a bitasset after a settlement has executed" );
+
+   // Are we changing the backing asset?
+   if( op.new_options.short_backing_asset != current_bitasset_data.options.short_backing_asset )
    {
-      FC_ASSERT(a.dynamic_asset_data_id(d).current_supply == 0);
-      FC_ASSERT(d.find_object(o.new_options.short_backing_asset));
+      FC_ASSERT(asset_obj.dynamic_asset_data_id(d).current_supply == 0,
+            "Cannot update a bitasset if there is already a current supply.");
+      FC_ASSERT(d.find_object(op.new_options.short_backing_asset),
+            "The short backing asset could not be found");
 
-      if( a.issuer == GRAPHENE_COMMITTEE_ACCOUNT )
+      const asset_object& new_backing_asset = op.new_options.short_backing_asset(d);
```

Comment:
Can combine this assignment with the `find_object` call above for better performance.